### ISSUE TITLE
Doing nativeImage.toPng() with the result of capturePage blocks the main thread for a few seconds

### ISSUE THREAD

**Issue by romaincointepas** [electron/electron#9445]
* Electron version: 1.6.9, 1.7.0
* Operating system: macOS 10.12.4

### Expected behavior

A non-blocking way to do the following (end goal is to save as png file):

```js
win.capturePage(function(image) {
  var png = image.toPng()
})
```

### Actual behavior

`image.toPng()` blocks the main process for about 3 seconds on the latest Macbook Pro when image is 2880x1800 (fullscreen retina display).

Would it be possible to either offer an async version of nativeImage.toPng() (and other similar methods) or add an option to `BrowserWindow.capturePage()` to return a PNG (or JPG/BMP) image directly as node buffer?

Every time someone needs to save a page screenshot, there is always going to be something blocking the thread (500ms-1s for smaller images, to way higher for fullscreen screenshot on low-end machines).

**Comment by countravioli** [comment]
To add some notes, I did some comparison testing against the toJpeg etc... methods and this is what I've noticed. Note: I'm capturing a 1280x720 size rect right now.

```
nativeImageInstance.toPng() //~ .5secs
nativeImageInstance.toJpeg(95) //~ .012secs
nativeImageInstance.toJpeg(100) //~ .018secs
nativeImageInstance.toBitmap() //~ .003secs
nativeImageInstance.getBitmap() //~ .001secs
```

I don't know if thats just because converting to PNGs is so much more difficult of an algorithm, just weighing in because this is definitely takes longer than I expected, which is pretty bad considering the blocking nature of it as the OP mentioned.
